### PR TITLE
Drop logging feature gate

### LINKED ISSUE #5277: Drop `Logging` feature gate in favor of gardenlet configuration
**How to categorize this issue?**

/area usability
/kind enhancement

**What would you like to be added**:
The `Logging` feature gate should be deprecated and removed. Instead, the logging feature should be enabled/disabled/configured via the gardenlet configuration.

**Why is this needed**:
The feature will never be promoted given that it's an optional feature, so having introduced it in the first place was the wrong approach. We already have logging-related configuration in the gardenlet config: https://github.com/gardener/gardener/blob/b822dfb5e86d418f541bb66dea4a6632c78f6fc5/example/20-componentconfig-gardenlet.yaml#L126-L137

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area usability
/kind enhancement

**What this PR does / why we need it**:
The Logging feature gate is removed now logging feature is enabled/disabled/configured via the gardenlet configuration.

**Which issue(s) this PR fixes**:
Fixes #5277

**Special notes for your reviewer**:
CC - @vlvasilev 

**Release note**:

```breaking operator
The Logging feature gate is removed, now logging is enabled/disabled/configured via the gardenlet configuration.
```
```breaking operator
A new field `Enabled` is introduced in `Logging` field of the gardenlet configuration to enable/disable logging. By default it is set to `false`.
```

### CODE REVIEW

File: docs/development/local_setup.md [hunk 1]
```diff
@@ -306,12 +306,6 @@ apiservice.apiregistration.k8s.io/v1alpha1.seedmanagement.gardener.cloud created
 apiservice.apiregistration.k8s.io/v1alpha1.settings.gardener.cloud created
 ```
 
-Optionally, you can switch off the `Logging` feature gate of Gardenlet to save resources:
-
-```bash
-sed -i -e 's/Logging: true/Logging: false/g' dev/20-componentconfig-gardenlet.yaml
-```
-
```
Comment (rfranzke) [inline docs/development/local_setup.md]: Instead of removing this, what about mentioning the new `logging.enabled` option in the config?

File: pkg/gardenlet/apis/config/v1alpha1/types.go [hunk 2]
```diff
@@ -469,6 +469,9 @@ type ShootNodeLogging struct {
 
 // Logging contains configuration for the logging stack.
 type Logging struct {
+	// Enabled is used to enable or disable logging stack for clusters.
+	// +optional
+	Enabled *bool
```
Comment (rfranzke) [inline pkg/gardenlet/apis/config/v1alpha1/types.go]: json/yaml tags are missing

File: pkg/operation/botanist/botanist.go [hunk 3]
```diff
@@ -232,3 +232,13 @@ func (b *Botanist) RequiredExtensionsReady(ctx context.Context) error {
 
 	return nil
 }
+
+// IsLoggingEnabled return true if the logging stack for clusters is enabled.
+func (b *Botanist) IsLoggingEnabled() bool {
```
Comment (rfranzke) [inline pkg/operation/botanist/botanist.go]: unit test?

Comment (rfranzke) [reply pkg/operation/botanist/botanist.go]: This should probably rather be a helper function (https://github.com/gardener/gardener/tree/master/pkg/gardenlet/apis/config/helper)

File: pkg/operation/botanist/logging.go [hunk 4]
```diff
@@ -162,8 +162,8 @@ func (b *Botanist) destroyShootNodeLogging(ctx context.Context) error {
 }
 
 func (b *Botanist) isShootNodeLoggingEnabled() bool {
-	if b.Shoot != nil && b.Shoot.IsLoggingEnabled() && b.Config != nil &&
-		b.Config.Logging != nil && b.Config.Logging.ShootNodeLogging != nil {
```
Comment (rfranzke) [inline pkg/operation/botanist/logging.go]: The `!= nil` checks seem still reasonable

**Review by rfranzke — CHANGES_REQUESTED** [review]

File: pkg/operation/botanist/logging.go (hunk 4 above)
Comment (acumino) [reply pkg/operation/botanist/logging.go]: Adding `!=nil` is not required but It will make things clear. I will add it. Thanks

File: docs/development/local_setup.md (hunk 1 above)
Comment (acumino) [reply docs/development/local_setup.md]: Addressed in 8c9ee3ad536361a54bdf56825cd1cb78a2bc8b72. PTAL

File: pkg/gardenlet/apis/config/v1alpha1/types.go (hunk 2 above)
Comment (acumino) [reply pkg/gardenlet/apis/config/v1alpha1/types.go]: Addressed in 8c9ee3ad536361a54bdf56825cd1cb78a2bc8b72.PTAL

File: pkg/operation/botanist/botanist.go (hunk 3 above)
Comment (acumino) [reply pkg/operation/botanist/botanist.go]: Addressed in 8c9ee3ad536361a54bdf56825cd1cb78a2bc8b72.PTAL

File: pkg/operation/botanist/logging.go (hunk 4 above)
Comment (acumino) [reply pkg/operation/botanist/logging.go]: Addressed in 8c9ee3ad536361a54bdf56825cd1cb78a2bc8b72. PTAL

**Review by rfranzke — DISMISSED** [review]
/lgtm

File: pkg/gardenlet/controller/shoot/seed_registration_control_test.go [hunk 5]
```diff
@@ -298,7 +297,7 @@ var _ = Describe("SeedRegistrationReconciler", func() {
 				})
 
 				It("should create the correct ManagedSeed resource, all custom", func() {
-					shoot := newShoot("true,disable-dns,disable-capacity-reservation,protected,invisible,minimumVolumeSize=20Gi,apiServer.replicas=2,apiServer.autoscaler.minReplicas=2,apiServer.autoscaler.maxReplicas=5,blockCIDRs=169.254.169.254/32,shootDefaults.pods=100.96.0.0/11,shootDefaults.services=100.64.0.0/13,backup.provider=gcp,backup.region=europe-north1,use-serviceaccount-bootstrapping,with-secret-ref,featureGates.Logging=false,resources.capacity.shoots=100,loadBalancerServices.annotations.foo=bar,ingress.controller.kind=nginx")
+					shoot := newShoot("true,disable-dns,disable-capacity-reservation,protected,invisible,minimumVolumeSize=20Gi,apiServer.replicas=2,apiServer.autoscaler.minReplicas=2,apiServer.autoscaler.maxReplicas=5,blockCIDRs=169.254.169.254/32,shootDefaults.pods=100.96.0.0/11,shootDefaults.services=100.64.0.0/13,backup.provider=gcp,backup.region=europe-north1,use-serviceaccount-bootstrapping,with-secret-ref,resources.capacity.shoots=100,loadBalancerServices.annotations.foo=bar,ingress.controller.kind=nginx")
```
Comment (ialidzhikov) [inline pkg/gardenlet/controller/shoot/seed_registration_control_test.go]: Can you preserve this test/scenario but with just another feature gate?

File: pkg/gardenlet/controller/managedseed/managedseed_valueshelper_test.go [hunk 6]
```diff
@@ -110,8 +110,7 @@ var _ = Describe("ValuesHelper", func() {
 				},
 			},
 			FeatureGates: map[string]bool{
-				string(features.Logging): true,
```
Comment (ialidzhikov) [inline pkg/gardenlet/controller/managedseed/managedseed_valueshelper_test.go:113]: Can you keep the existing test/scenario but just use a different feature gate? This tests seems to be related to merging of gardenlet configs, so obviously we don't want to drop a test case.

**Review by ialidzhikov — CHANGES_REQUESTED** [review]

File: pkg/gardenlet/controller/shoot/seed_registration_control_test.go (hunk 5 above)
Comment (acumino) [reply pkg/gardenlet/controller/shoot/seed_registration_control_test.go]: For this case, Since the feature gate was set to false, and by default logging is disabled, I guess this test scenario is preserved. PTAL?

File: pkg/operation/botanist/logging.go [hunk 7]
```diff
@@ -46,7 +47,7 @@ func (b *Botanist) DeploySeedLogging(ctx context.Context) error {
 		lokiEnabled = *b.Config.Logging.Loki.Enabled
 	}
 
-	if !b.Shoot.IsLoggingEnabled() || !lokiEnabled {
+	if !b.Shoot.IsLoggingEnabled() || gardenlethelper.IsLoggingEnabled(b.Config) || !lokiEnabled {
```
Comment (vlvasilev) [inline pkg/operation/botanist/logging.go]:
```suggestion
	if !b.Shoot.IsLoggingEnabled() || !gardenlethelper.IsLoggingEnabled(b.Config) || !lokiEnabled {
```

File: pkg/operation/botanist/logging.go [hunk 8]
```diff
@@ -163,7 +164,8 @@ func (b *Botanist) destroyShootNodeLogging(ctx context.Context) error {
 
 func (b *Botanist) isShootNodeLoggingEnabled() bool {
 	if b.Shoot != nil && b.Shoot.IsLoggingEnabled() && b.Config != nil &&
-		b.Config.Logging != nil && b.Config.Logging.ShootNodeLogging != nil {
+		b.Config.Logging != nil && gardenlethelper.IsLoggingEnabled(b.Config) &&
+		b.Config.Logging.ShootNodeLogging != nil {
```
Comment (vlvasilev) [inline pkg/operation/botanist/logging.go]: Maybe here we should consider if the Loki is disabled because without it there is no sense to have `shootnode` logging at all.
```suggestion
		b.Config.Logging.ShootNodeLogging != nil  && 
		b.Config.Logging.Loki != nil &&
		b.Config.Logging.Loki.Enabled != nil &&
		*b.Config.Logging.Loki.Enabled {
```

File: pkg/gardenlet/controller/managedseed/managedseed_valueshelper_test.go (hunk 6 above)
Comment (acumino) [reply pkg/gardenlet/controller/managedseed/managedseed_valueshelper_test.go:113]: Addressed in 6195a6338312c0bd41c0af2ca8a58ed1c7516dfc. PTAL

File: pkg/operation/care/health.go [hunk 9]
```diff
@@ -292,7 +290,14 @@ func (h *Health) checkControlPlane(
...
 
-	if gardenletfeatures.FeatureGate.Enabled(features.Logging) {
+	var loggingEnabled = false
+	if h.gardenletConfiguration != nil &&
+		h.gardenletConfiguration.Logging != nil &&
+		h.gardenletConfiguration.Logging.Enabled != nil {
+		loggingEnabled = *h.gardenletConfiguration.Logging.Enabled
+	}
+
+	if loggingEnabled {
```
Comment (vlvasilev) [inline pkg/operation/care/health.go:290]:
```suggestion
	if exitCondition, err := checker.CheckLoggingControlPlane(h.shoot.SeedNamespace, 
            h.shoot.IsLoggingEnabled() && loggingEnabled , lokiEnabled, condition, 
             seedStatefulSetLister); err != nil || exitCondition != nil {
			return exitCondition, err
	}
```
`h.shoot.IsLoggingEnabled()` and  `h.shoot.Purpose == gardencorev1beta1.ShootPurposeTesting` are the same.

**Review by vlvasilev — CHANGES_REQUESTED** [review]

File: pkg/operation/botanist/logging.go (hunk 7 above)
Comment (acumino) [reply pkg/operation/botanist/logging.go]: Thanks, I have already made the change in 2c706bc5bb102f9dee3a717b07faa38e725c14af.

File: pkg/operation/care/health.go (hunk 9 above)
Comment (acumino) [reply pkg/operation/care/health.go:290]: Thanks, I have already made the change in 2c706bc5bb102f9dee3a717b07faa38e725c14af.

File: pkg/gardenlet/controller/shoot/seed_registration_control_test.go (hunk 5 above)
Comment (ialidzhikov) [reply pkg/gardenlet/controller/shoot/seed_registration_control_test.go]: This test is not about the Logging feature gate value and its default. It is rather about the the parsing of feature gates in the `use-as-seed` annotation - Logging is just an example feature gate that is used for the test purpose to verify that a feature gate gets parsed and gets set in the ManagedSeed spec.

If I go and introduce the following regression:
```diff
diff --git a/pkg/apis/core/v1beta1/helper/helper.go b/pkg/apis/core/v1beta1/helper/helper.go
index 4544ae83c..59f31c9e7 100644
--- a/pkg/apis/core/v1beta1/helper/helper.go
+++ b/pkg/apis/core/v1beta1/helper/helper.go
@@ -312,7 +312,6 @@ func parseShootedSeed(annotation string) (*ShootedSeed, error) {

        shootedSeed := ShootedSeed{
                LoadBalancerServicesAnnotations: parseShootedSeedLoadBalancerServicesAnnotations(settings),
-               FeatureGates:                    parseShootedSeedFeatureGates(settings),
        }

        apiServer, err := parseShootedSeedAPIServer(settings)
```

this test (with the change from this PR) will no longer fail (catch an issue).

File: pkg/operation/botanist/logging.go (hunk 8 above)
Comment (acumino) [reply pkg/operation/botanist/logging.go]: It's not required cause if Loki is not enabled it will stop [here](https://github.com/gardener/gardener/blob/04d2b3f459d198e8db0ab57180ca2fea18e84da9/pkg/operation/botanist/logging.go#L50-L52) only.

File: pkg/gardenlet/controller/shoot/seed_registration_control_test.go (hunk 5 above)
Comment (acumino) [reply pkg/gardenlet/controller/shoot/seed_registration_control_test.go]: Thanks for the explanation. Updated it e7c61696dfb8caf9f93cd06f3bf4df92b10dcd09

File: pkg/gardenlet/controller/managedseed/managedseed_valueshelper_test.go (hunk 6 above)
Comment (acumino) [reply pkg/gardenlet/controller/managedseed/managedseed_valueshelper_test.go:113]: Here we don't need it as there is already one feature gate present.

**Comment by acumino** [comment]
@ialidzhikov Can this be included in v1.40?

File: pkg/operation/botanist/logging.go (hunk 8 above)
Comment (vlvasilev) [reply pkg/operation/botanist/logging.go]: The link refers to the logic which deploys the Loki in the shoot controlplane.
[This](https://github.com/gardener/gardener/blob/45da551bad81c5a3978cf2186b37284d0399dc50/pkg/operation/botanist/operatingsystemconfig.go#L61) is the logic which install the promtail on the shoot nodes

File: pkg/operation/care/health.go (hunk 9 above)
Comment (vlvasilev) [reply pkg/operation/care/health.go:290]: No, you haven't. Just remove `h.shoot.Purpose == gardencorev1beta1.ShootPurposeTesting` from the  `checker.CheckLoggingControlPlane()` call. Then replace it with `h.shoot.IsLoggingEnabled() && loggingEnabled` and then you could remove the first `if` enclosure.

File: pkg/operation/botanist/logging.go (hunk 8 above)
Comment (acumino) [reply pkg/operation/botanist/logging.go]: Previously this check was not there, I am not sure was it done intentionally in #3813.

Comment (vlvasilev) [reply pkg/operation/botanist/logging.go]: This is because it was omitted by mistake. But ok you could simply ignore it. I will fix it in another PR

File: pkg/gardenlet/controller/shoot/seed_registration_control_test.go [hunk 10]
```diff
@@ -298,7 +298,7 @@ var _ = Describe("SeedRegistrationReconciler", func() {
 				})
 
 				It("should create the correct ManagedSeed resource, all custom", func() {
-					shoot := newShoot("true,disable-dns,disable-capacity-reservation,protected,invisible,minimumVolumeSize=20Gi,apiServer.replicas=2,apiServer.autoscaler.minReplicas=2,apiServer.autoscaler.maxReplicas=5,blockCIDRs=169.254.169.254/32,shootDefaults.pods=100.96.0.0/11,shootDefaults.services=100.64.0.0/13,backup.provider=gcp,backup.region=europe-north1,use-serviceaccount-bootstrapping,with-secret-ref,featureGates.Logging=false,resources.capacity.shoots=100,loadBalancerServices.annotations.foo=bar,ingress.controller.kind=nginx")
+					shoot := newShoot("true,disable-dns,disable-capacity-reservation,protected,invisible,minimumVolumeSize=20Gi,apiServer.replicas=2,apiServer.autoscaler.minReplicas=2,apiServer.autoscaler.maxReplicas=5,blockCIDRs=169.254.169.254/32,shootDefaults.pods=100.96.0.0/11,shootDefaults.services=100.64.0.0/13,backup.provider=gcp,backup.region=europe-north1,use-serviceaccount-bootstrapping,with-secret-ref,featureGates.SeedChange=false,resources.capacity.shoots=100,loadBalancerServices.annotations.foo=bar,ingress.controller.kind=nginx")
```
Comment (ialidzhikov) [inline pkg/gardenlet/controller/shoot/seed_registration_control_test.go]: Nit: `SeedChange` is a feature gate of gardener-apiserver, so it is conceptually wrong to use it in unit test related to gardenlet feature gates.

```suggestion
					shoot := newShoot("true,disable-dns,disable-capacity-reservation,protected,invisible,minimumVolumeSize=20Gi,apiServer.replicas=2,apiServer.autoscaler.minReplicas=2,apiServer.autoscaler.maxReplicas=5,blockCIDRs=169.254.169.254/32,shootDefaults.pods=100.96.0.0/11,shootDefaults.services=100.64.0.0/13,backup.provider=gcp,backup.region=europe-north1,use-serviceaccount-bootstrapping,with-secret-ref,featureGates.ReversedVPN=false,resources.capacity.shoots=100,loadBalancerServices.annotations.foo=bar,ingress.controller.kind=nginx")
```

File: pkg/gardenlet/controller/shoot/seed_registration_control_test.go [hunk 11]
```diff
@@ -325,7 +325,7 @@ var _ = Describe("SeedRegistrationReconciler", func() {
 										},
 									},
 									FeatureGates: map[string]bool{
-										string(features.Logging): false,
+										string(features.SeedChange): false,
```
Comment (ialidzhikov) [inline pkg/gardenlet/controller/shoot/seed_registration_control_test.go]:
```suggestion
										string(features.ReversedVPN): false,
```

File: pkg/operation/shoot/shoot.go [hunk 12]
```diff
@@ -454,7 +454,7 @@ func (s *Shoot) IPVSEnabled() bool {
 
 // IsLoggingEnabled return true if the Shoot controlplane logging is enabled
 func (s *Shoot) IsLoggingEnabled() bool {
```
Comment (ialidzhikov) [inline pkg/operation/shoot/shoot.go]: Previously `IsLoggingEnabled` was checking both (the Shoot purpose and the feature gate value). Now it is checking only the purpose and now each and every usage has to call both `shoot.IsLoggingEnabled` and `gardenlethelper.IsLoggingEnabled`, right? Does it make sense to add the `gardenlethelper.IsLoggingEnabled` check to this func and preserve the old semantics of this func?

File: pkg/operation/seed/seed.go [hunk 13]
```diff
@@ -513,13 +513,18 @@ func RunReconcileSeedFlow(
...
 		filters                           = strings.Builder{}
 		parsers                           = strings.Builder{}
 		fluentBitConfigurationsOverwrites = map[string]interface{}{}
 		lokiValues                        = map[string]interface{}{}
 	)
 
+	if loggingConfig != nil &&
+		loggingConfig.Enabled != nil {
+		loggingEnabled = *loggingConfig.Enabled
+	}
```
Comment (ialidzhikov) [inline pkg/operation/seed/seed.go]: Can't we use `gardenlethelper.IsLoggingEnabled` directly? Why we repeat the same nil checks here as well?

File: pkg/operation/care/health.go [hunk 14]
```diff
@@ -292,7 +290,14 @@ func (h *Health) checkControlPlane(
 		lokiEnabled = *h.gardenletConfiguration.Logging.Loki.Enabled
 	}
 
-	if gardenletfeatures.FeatureGate.Enabled(features.Logging) {
+	var loggingEnabled = false
+	if h.gardenletConfiguration != nil &&
+		h.gardenletConfiguration.Logging != nil &&
+		h.gardenletConfiguration.Logging.Enabled != nil {
+		loggingEnabled = *h.gardenletConfiguration.Logging.Enabled
```
Comment (ialidzhikov) [inline pkg/operation/care/health.go]: Can't we use `gardenlethelper.IsLoggingEnabled` directly? Why we repeat the same nil checks here as well?

File: pkg/gardenlet/apis/config/helper/helpers.go [hunk 15]
```diff
@@ -91,3 +91,12 @@ func ConvertGardenletConfigurationExternal(obj runtime.Object) (*configv1alpha1.
 	}
 	return result, nil
 }
+
+// IsLoggingEnabled return true if the logging stack for clusters is enabled.
+func IsLoggingEnabled(c *config.GardenletConfiguration) bool {
+	if c != nil && c.Logging != nil &&
```
Comment (ialidzhikov) [inline pkg/gardenlet/apis/config/helper/helpers.go:97]: Is there really a case where the passed GardenletConfiguration can be nil?

File: docs/deployment/configuring_logging.md [hunk 16]
```diff
@@ -2,12 +2,10 @@
 
 # Enable the Logging
 
-Logging feature gate must be enabled in order to install the Gardener Logging stack.
-
-In the Gardenlet configuration add:
+To enable logging, in the Gardenlet configuration add:
```
Comment (ialidzhikov) [inline docs/deployment/configuring_logging.md]:
```suggestion
In order to install the Gardener logging stack the `logging.enabled` configuration option has to be enabled in the Gardenlet configuration:
```

File: pkg/gardenlet/controller/managedseed/managedseed_valueshelper_test.go (hunk 6 above)
Comment (ialidzhikov) [reply pkg/gardenlet/controller/managedseed/managedseed_valueshelper_test.go:113]: I don't think so. So in the master branch of this test we have a feature gate (Logging) that is enabled in the parent gardenlet config and the same feature gate (Logging) disabled in the gardenlet config. And the test verifies that in the merged gardenlet config the feature gate value from the gardenlet config gets respected (parent config gets overwritten). With this change we drop exactly this case (there is another feature gate but it is with values true in both configs).

**Review by ialidzhikov — CHANGES_REQUESTED** [review]

File: pkg/operation/shoot/shoot.go (hunk 12 above)
Comment (vlvasilev) [reply pkg/operation/shoot/shoot.go]: These two functions are related to different objects but if one of them is set to "false" then the Logging feature for the shoot is disabled.  In this case, we could make one utility function that accepts both the `shoot` object and the `gardenlet` configuration.

**Review by vlvasilev — CHANGES_REQUESTED** [review]

File: pkg/gardenlet/apis/config/helper/helpers.go (hunk 15 above)
Comment (acumino) [reply pkg/gardenlet/apis/config/helper/helpers.go:97]: I have not checked, but previously this condition was being checked so maybe yes.

File: pkg/operation/shoot/shoot.go (hunk 12 above)
Comment (acumino) [reply pkg/operation/shoot/shoot.go]: Probably keeping it separate is fine, this makes things easy to understand that both the things i.e. `shoot.IsLoggingEnabled` and `gardenlethelper.IsLoggingEnabled` should be enabled for logging stack. WDYT? @ialidzhikov @vlvasilev

Comment (vlvasilev) [reply pkg/operation/shoot/shoot.go]: I am ok with the separated functions. I don't think that the shoot has to know about the `gardenlet` configuration.
But if  @ialidzhikov suggests were to put the common function I will be ok to combine them.

**Review by vlvasilev — DISMISSED** [review]
/lgtm

**Review by rfranzke — DISMISSED** [review]
/lgtm

File: pkg/gardenlet/apis/config/helper/helpers.go (hunk 15 above)
Comment (ialidzhikov) [reply pkg/gardenlet/apis/config/helper/helpers.go:97]: Can you add a test to cover the case when the passed GardenletConfiguration is nil?

File: pkg/gardenlet/apis/config/helper/helpers.go [hunk 17]
```diff
@@ -91,3 +91,21 @@ func ConvertGardenletConfigurationExternal(obj runtime.Object) (*configv1alpha1.
...
+		c.Logging.Enabled != nil {
+		return *c.Logging.Enabled
+	}
+	return false
+}
+
+// IsLokiEnabled return true if the loki is enabled
+func IsLokiEnabled(c *config.GardenletConfiguration) bool {
+	if c != nil && c.Logging != nil &&
```
Comment (ialidzhikov) [inline pkg/gardenlet/apis/config/helper/helpers.go:106]: Can you add a test to cover the case when the passed GardenletConfiguration is nil?

File: pkg/operation/shoot/shoot.go (hunk 12 above)
Comment (ialidzhikov) [reply pkg/operation/shoot/shoot.go]: I would rather vote to combine in 1 function:
- it is error-prone with 2 funcs, one can miss to call one of funcs
- it is cleaner and less code duplication

File: pkg/gardenlet/apis/config/helper/helpers.go (hunk 17 above)
Comment (acumino) [reply pkg/gardenlet/apis/config/helper/helpers.go:106]: Can you please check `helpers_test.go`. Is something else required? I think those tests cover this scenario.

Comment (ialidzhikov) [reply pkg/gardenlet/apis/config/helper/helpers.go:106]: I checked and I don't see a test that invokes `IsLokiEnabled` with a nil GardenletConfiguration.

File: pkg/operation/shoot/shoot.go (hunk 12 above)
Comment (acumino) [reply pkg/operation/shoot/shoot.go]: Updated it in ab9857feb0bbb537a44aaf636295579ed447fa99. PTAL

**Review by ialidzhikov — DISMISSED** [review]
lgtm from high-level look (I haven't played with the change locally).

/squash

File: docs/deployment/feature_gates.md [hunk 18]
```diff
@@ -59,6 +58,8 @@ The following tables are a summary of the feature gates that you can set on diff
 | DisallowKubeconfigRotationForShootInDeletion | `true`  | `Beta`    | `1.32` | `1.35` |
 | DisallowKubeconfigRotationForShootInDeletion | `true`  | `GA`      | `1.36` |        |
 | DisallowKubeconfigRotationForShootInDeletion |         | `Removed` | `1.38` |        |
+| Logging                                      | `false` | `Alpha`   | `0.13` | `1.39` |
```
Comment (ialidzhikov) [inline docs/deployment/feature_gates.md]: Nit

```suggestion
| Logging                                      | `false` | `Alpha`   | `0.13` | `1.40` |
```

**Review by rfranzke — APPROVED** [review]
/cc @plkokanov as release responsible for this breaking change
